java/src/main/java/com/google/appengine/tools/mapreduce/impl/util/EntityUtils.java

```diff
@@ -0,0 +1,67 @@
+package com.google.appengine.tools.mapreduce.impl.util;
+
+import com.google.cloud.datastore.*;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Copyright 2020 - Worklytics, Co.
+ */
+
+/**
+ * class to wrap Entity with accessors that are safe for missing properties
+ *
+ * TODO: unify w com.google.appengine.tools.pipeline.impl.util.EntityUtils, when merge the two projects
+ */
+public class EntityUtils {
+  final static String IS_SHARDED_PROPERTY = "isSharded";
+
+  public static Key getKey(Entity entity, String propertyName) {
```

Comment:
Probably better with `Optional`, but that could impact in more places, so maybe for a 2nd iteration